client/src/lib.rs

```diff
@@ -229,42 +249,69 @@ impl<C: Deref<Target = impl Signer> + Clone> Program<C> {
 
     /// Returns a request builder.
     pub fn request(&self) -> RequestBuilder<C> {
+        #[cfg(not(feature = "rpc-client"))]
+        return RequestBuilder::from(
+            self.program_id,
+            self.cfg.cluster.url(),
+            self.cfg.payer.clone(),
+            self.cfg.options,
+            #[cfg(not(feature = "async"))]
+            self.rt.handle(),
+        );
```

Comment:
Same unnecessary duplication here, we should be able to just add the feature flag check above the `&self.async_rpc_client` argument.